### PR TITLE
Turn on pull mode by default

### PR DESCRIPTION
**Pull request by hidenori-shinohara**
# Description

Resolves https://github.com/stellar/stellar-core/issues/3572

This turns on pull mode by default. Note that node operators can still turn it off through the config file. Furthermore, this bumps the overlay protocol version.

# Checklist
- [x] Reviewed the [contributing](https://github.com/stellar/stellar-core/blob/master/CONTRIBUTING.md#submitting-changes) document
- [x] Rebased on top of master (no merge commits)
- [x] Ran `clang-format` v8.0.0 (via `make format` or the Visual Studio extension)
- [x] Compiles
- [x] Ran all tests
- [x] If change impacts performance, include supporting evidence per the [performance document](https://github.com/stellar/stellar-core/blob/master/performance-eval/performance-eval.md)

### CODE REVIEW

**Review by marta-lokhova — CHANGES_REQUESTED** [review]
thanks. we also need to make sure that minimum overlay version is up-to-date

**Comment by marta-lokhova** [comment]
note: we should merge this change last, before closing the tree, so we can do perf run before/after with all the other changes already in

File: src/main/Config.cpp [hunk 1]
```diff
@@ -127,7 +127,7 @@ Config::Config() : NODE_SEED(SecretKey::random())
 
     MAXIMUM_LEDGER_CLOSETIME_DRIFT = 50;
 
-    OVERLAY_PROTOCOL_MIN_VERSION = 21;
+    OVERLAY_PROTOCOL_MIN_VERSION = 22;
```
Comment (MonsieurNicolas) [inline src/main/Config.cpp:130]: we can actually bump to 23 that corresponds to 19.3.0 released in July (this is in line with our release policy of 90 days)

**Review by MonsieurNicolas — CHANGES_REQUESTED** [review]

File: src/main/Config.cpp (hunk 1 above)
Comment (hidenori-shinohara) [reply src/main/Config.cpp]: Updated!

**Comment by MonsieurNicolas** [comment]
r+ 83da984d3dc05eb67ed50b0045de0045c697dcac

**Pull request merged**